Update dependencies, unbreak examples

est31 opened the pull request:
This PR updates the glutin dependency to 0.20, as well as some other dependencies, so that cargo-outdated doesn't complain any more. Also, it issues a new release and unbreaks the shadow mapping example which previously failed to compile.

est31 commented:
@mitchmindtree as there seems to be no feedback from @tomaka, could you review and merge this PR?

mitchmindtree commented:
This looks fine to me, though it would be nice if @ZeGentzy could give the thumbs up in case we've missed anything?

@est31 if there's no reply within the next day or two I'd be happy to merge :+1:

ZeGentzy commented on src/backend/glutin/mod.rs:
```diff
@@ -135,11 +135,11 @@ impl Display {
             let gl_window = self.gl_window.borrow();
             let context_builder = context_builder.with_shared_lists(gl_window.context());
-            try!(glutin::GlWindow::new(window_builder, context_builder, events_loop))
+            try!(glutin::WindowedContext::new_windowed(window_builder, context_builder, events_loop))
```
Nit: Might as well change `try!(...)` to `...?`.

ZeGentzy commented on examples/gpgpu.rs:
```diff
@@ -2,11 +2,16 @@
+        width: 800.0,
+        height: 600.0,
+    };
+    let context = glutin::Context::new_headless(&event_loop, context_builder, size).unwrap();
```
Nit: go `glutin::ContextBuilder::new().build_headless(...).unwrap()`, instead of `glutin::Context::new_headless(..).unwrap()`.

ZeGentzy commented on examples/manual-creation.rs:
```diff
@@ -30,12 +30,12 @@ fn main() {
     let window = glutin::WindowBuilder::new();
     let context = glutin::ContextBuilder::new();
-    let gl_window = Rc::new(glutin::GlWindow::new(window, context, &events_loop).unwrap());
+    let gl_window = Rc::new(glutin::WindowedContext::new_windowed(window, context, &events_loop).unwrap());
```
Nit: go `glutin::ContextBuilder::new().new_windowed(...).unwrap()`, instead of `glutin::Context::new_windowed(..).unwrap()`.

ZeGentzy commented on examples/shadow_mapping.rs:
```diff
@@ -5,15 +5,19 @@ extern crate cgmath;
-        .with_dimensions(win_size.0, win_size.1)
+        .with_dimensions(win_size)
         .with_title("Shadow Mapping");
     let context = glutin::ContextBuilder::new().with_vsync(true);
```
Nit: What you have here isn't a context, but a context builder. The name `context` would be slightly misleading. How about something short, like `cb`, or something not so short, like `context_builder`?

est31 replied:
Yes, but its not this PR which has added the code. Maybe something for a later PR that focuses on refactoring?

ZeGentzy commented on src/backend/glutin/mod.rs:
```diff
@@ -67,44 +67,44 @@ impl Display {
     ) -> Result<Self, DisplayCreationError>
     {
-        let gl_window = try!(glutin::GlWindow::new(window_builder, context_builder, events_loop));
+        let gl_window = try!(glutin::WindowedContext::new_windowed(window_builder, context_builder, events_loop));
```
Nit: Might as well change `try!(...)` to `...?`.

ZeGentzy commented:
As an minor note, glutin v0.21.0-rc2 is out. Could you update to that? You got to modify the version in te `Cargo.toml` manually, as it's a release candidate.

est31 commented:
I've pushed a commit that addresses some of the nits.

> glutin v0.21.0-rc2 is out. Could you update to that?

Sure, I'd like to do that in a followup PR. I have some further minor changes in mind that are less related to the version update. Then we have one release with glutin 0.20, and once 0.21 is out, we can release a new glium version again.

est31 commented:
Merge?
